forge/csrc/passes/split_graph.cpp

```diff
@@ -333,26 +332,147 @@ std::unique_ptr<Graph> extract_backward_graph(
     return bwd_graph;
 }
 
+std::unique_ptr<Graph> extract_optimizer_graph(
+    const Graph *graph, const Graph *fwd_graph, const std::vector<graphlib::Node *> &topo)
+{
+    auto opt_graph = std::make_unique<Graph>(tt::graphlib::IRLevel::IR_TT_FORGE, "optimizer");
+    opt_graph->set_training(graph->training());
+
+    // Adding all trainable params (which require gradients) from the forward graph as inputs to the optimizer graph.
+    std::vector<graphlib::NodeId> opt_module_outputs;
+    for (auto param : graph->get_parameter_nodes())
+    {
+        if (param->as<graphlib::InputNode>()->requires_grad())
```

Comment:
Do we have a test covering cases where not all parameters are trainable? 

I'm mostly interested in e2e test which proves fine-tuning case with mixed required_grad options.

Reply:
Sure. I think we should have both. One simple case for mixed trainable params, and a specific one that targets the uniqueness of the LoRA case. 

For example, if a simple mix grad test fails, it's easier to debug as it has a single purpose compared to LoRA.